documentation/versioning.md

```diff
@@ -0,0 +1,118 @@
+- Services can version & be retired independently from each other. This allows services to be agile and removes coordination between sub-teams.  
+- Customers consciously select and target specific service versions.
+    - This is important for the public cloud customers as all datacenters may not have the same service versions deployed. 
+    - It is also important for Azure Stack, government cloud, and other air-gapped/non-public cloud customers which also may not have the same service versions deployed.  
+
+## An Azure Resource Provider and its 1+ Azure Services
+
+A single Azure Resource Provider (RP) has an RP namespace. This RP namespace can contain 1+ control-plane services and 0+ data-plane service; each service versions as described above . Today, we want the GitHub specs repo to be structured as follows for control-plane & data-plane contracts:
+```
+specification/
+└── <orgName1>/	ß NOTE: Today this has no clear definition (Org name? Service name?)
+    ├── cspell.yaml
+    └── resource-manager/
+        ├── readme.md			ß NOTE: For ARM schema validation; see bullet #2 below
+        └── <RPNamespace>/		ß NOTE: Control-plane only (not data-plane) 
+            └── <ServiceName1>/	ß CX-facing service name; each version gets Docs & SDK package
+                ├── tspconfig.yaml	ß TypeSpec files go here (examples below)
+                ├── main.tsp
+                ├── models.tsp
+                ├── readme.md		ß autorest readme with YAML blocks
+                └── examples/		ß TypeSpec example folder
+                    └── <api-version>/	ß One folder per service version described in TypeSpec
+                        └── <example .json files> 
+                └── preview/ and stable/
+                    └── <api-version>/	ß One folder per service version described in Swagger
+                        ├── <swagger .json files>
+                        └── examples/	ß Swagger example folder
+                            └── <example .json files>
+            └── <ServiceName2>/	// CX-facing service name; contents identical to above structure
+    └── data-plane/
+        └── <ServiceName3>/	// CX-facing service name; contents identical to above structure
+```
+We don’t want internal teams or customers to think about the orgName, we want customers to think that Azure has RPNamespaces that are used to manage services (both with customer-facing names) where each service has versions (some preview and some stable), & each version has its own API contract, documentation, and (beta/GA) SDK package(s).   
+
+To enforce these rules, Here're some tweaks to the repo structure:
+
+1.	Force the folder immediately under specification to be the RPNamespace name (replacing orgName). EngSys should validate that this folder name matches the RPnamespace name currently under the resource-manager folder and that there are no files in this folder and that the only subfolders are resource-manager and data-plane. 
```

Comment:
@JeffreyRichter I wonder if we can delete this entry? As it would be misleading to services team and this is not what we currently do.